### PR TITLE
add Cancel Interaction RPC

### PR DESCRIPTION
**Pull request by iCollin**
Implements https://github.com/smartdevicelink/rpc_spec/issues/122

### CODE REVIEW

File: MOBILE_API.xml [hunk 1]
```diff
@@ -7311,6 +7312,46 @@
         </param>
     </function>
 
+    <function name="CancelInteraction" functionID="CancelInteractionID" messagetype="request" since="6.0">
```
Comment (jacobkeeler) [inline MOBILE_API.xml:7337]: Should probably include a description of the RPC, even though it wasn't included in the proposal

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: MOBILE_API.xml (hunk 1 above)
Comment (iCollin) [reply MOBILE_API.xml:7337]: added in https://github.com/smartdevicelink/rpc_spec/pull/178/commits/586b711f1f39a93f882edd0993735dd809162f31

**Review by jacobkeeler — APPROVED** [review]

**Review by theresalech — CHANGES_REQUESTED** [review]
I think you also need to add `cancelID` param to the following functions:
- `PerformInteraction`
- `Alert`
- `ScrollableMessage`
- `Slider`

See the "Additional Changes" section of [the proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0184-cancel-interaction.md) for details.

**Comment by iCollin** [comment]
> I think you also need to add `cancelID` param to the following functions...

Added in https://github.com/smartdevicelink/rpc_spec/pull/178/commits/916367cd44e34a7f06953e3792ed16fb18ef33e0

**Comment by theresalech** [comment]
@iCollin changes look good!  Please resolve merge conflicts and I'll approve the PR, thanks!

**Comment by theresalech** [comment]
Determined we will fix merge conflicts just before merge takes place.

**Review by theresalech — APPROVED** [review]

File: MOBILE_API.xml [hunk 2]
```diff
@@ -4770,6 +4771,12 @@
             <description>See LayoutMode.</description>
         </param>
         
+        <param name="cancelID" type="Integer" mandatory="false" since="6.0">
+            <description>
+                An ID for this specific alert to allow cancellation through the `CancelInteraction` RPC.
```
Comment (joeygrover) [inline MOBILE_API.xml]:
```suggestion
                An ID for this specific PerformInteraction to allow cancellation through the `CancelInteraction` RPC.
```

File: MOBILE_API.xml [hunk 3]
```diff
@@ -6117,6 +6130,11 @@
                 If omitted on supported displays, only the system defined "Close" SoftButton will be displayed.
             </description>
         </param>
+        <param name="cancelID" type="Integer" mandatory="false" since="6.0">
+            <description>
+                An ID for this specific alert to allow cancellation through the `CancelInteraction` RPC.
```
Comment (joeygrover) [inline MOBILE_API.xml]:
```suggestion
                An ID for this specific ScrollableMessage to allow cancellation through the `CancelInteraction` RPC.
```

File: MOBILE_API.xml [hunk 4]
```diff
@@ -6171,6 +6189,11 @@
                 If omitted, the value is set to 10000.
             </description>
         </param>
+        <param name="cancelID" type="Integer" mandatory="false" since="6.0">
+            <description>
+                An ID for this specific alert to allow cancellation through the `CancelInteraction` RPC.
```
Comment (joeygrover) [inline MOBILE_API.xml]:
```suggestion
                An ID for this specific Slider to allow cancellation through the `CancelInteraction` RPC.
```

**Pull request merged**